### PR TITLE
MET-6058 Introduce new tombstone dao

### PR DESCRIPTION
**Pull request by stzanakis**

### CODE REVIEW

**Review by jeortizquan — COMMENTED** [review]
Some suggestions are made.

File: metis-indexing/src/main/java/eu/europeana/indexing/IndexerFactory.java [hunk 1]
```diff
@@ -60,7 +74,11 @@ public IndexerFactory(IndexerConnectionSupplier connectionProviderSupplier) {
    */
   public Indexer getIndexer()
       throws SetupRelatedIndexingException, IndexerRelatedIndexingException {
-    return new IndexerImpl(connectionProviderSupplier.get());
+    try {
```
[the remark this answers is not in the document]
Comment (stzanakis) [reply metis-indexing/src/main/java/eu/europeana/indexing/IndexerFactory.java:77]: Done, I guess

File: metis-indexing/src/main/java/eu/europeana/indexing/FullBeanPublisher.java [hunk 2]
```diff
@@ -183,6 +187,8 @@ private void publish(RdfWrapper rdf, Date recordDate, List<String> datasetIdsToR
     final FullBeanImpl savedFullBean = publishToMongo(recordDate, fullBean, fullBeanPreprocessor,
         recordsForRedirection);
 
+    //Around here we need to use th new tombstone record dao.
```
[the remark this answers is not in the document]
Comment (stzanakis) [reply metis-indexing/src/main/java/eu/europeana/indexing/FullBeanPublisher.java:190]: We won't forget, don't worry :D 
I am actually not completely sure where the code exactly will be yet.

File: metis-indexing/src/main/java/eu/europeana/indexing/ClientsConnectionProvider.java [hunk 3]
```diff
@@ -1,62 +1,65 @@
...
-  private final SolrClient solrClient;
+  /**
+   * Constructor.
+   *
+   * @param recordDao The Mongo dao to be used. Cannot be null.
+   * @param tombstoneRecordDao The Mongo tombstone dao to be used. Cannot be null.
+   * @param recordRedirectDao The record redirect dao.
+   * @param solrClient The Solr client to be used. Cannot be null.
+   */
+  public ClientsConnectionProvider {
```
[the remark this answers is not in the document]
Comment (stzanakis) [reply metis-indexing/src/main/java/eu/europeana/indexing/ClientsConnectionProvider.java:25]: Done.

File: metis-indexing/src/main/java/eu/europeana/indexing/ClientsConnectionProvider.java
[the remark this answers is not in the document]
Comment (stzanakis) [reply metis-indexing/src/main/java/eu/europeana/indexing/ClientsConnectionProvider.java:1]: Sure, added

**Pull request merged**